File: sistema_java/codigos/oficinapoo/Especialista.java
package oficinapoo;

public class Especialista extends Pessoa
{
    public Assistente assistente;
    public Agenda agenda;
    
    //construtores
    public Especialista()
    {
        super("", "", "12345678900", "33988888888");
        assistente = new Assistente();
        agenda = new Agenda();   
    }
    
    public Especialista(String nome, String endereco, String cpf, String telefone)
    {
        super(nome, endereco, cpf, telefone);
        assistente = new Assistente();
        agenda = new Agenda();
    }
    
    public Especialista(Assistente assistente, Agenda agenda,String nome, String endereco, String cpf, String telefone)
    {
        super(nome, endereco, cpf, telefone);
	this.assistente = new Assistente(assistente);
        this.agenda = new Agenda(agenda);
    }
    
     public Especialista(Especialista especialista)
    {
        super(especialista.getNome(), especialista.getEndereco(), especialista.getCpf(), especialista.getTelefone());
	this.assistente = new Assistente(especialista.assistente);
        this.agenda = new Agenda(especialista.agenda);
    }

    public String imprimirAgenda()//imprime toda a agenda de um Especialista
    {
        String leitura = "";
	leitura = "Especialista:" + getNome()+ " CPF:"+ getCpf() + "\n\n" + agenda.toString();
        return leitura;
    }
    
    //sobrecarga do operador print
    public String toString()
    {
        String retorno = "";
        retorno = retorno + super.toString() +"\n";
        retorno = retorno + "\nAssistente Do Especialista:\nNome: " + assistente.getNome() 
                + " CPF: " + assistente.getCpf();
        
        return retorno;
    }

}

File: sistema_cpluplus/codigos/Especialista.cpp
#include "Especialista.h"
#include <iostream>//entrada e saida


using namespace std;

//construtor
Especialista::Especialista(Assistente& assistente, Agenda& agenda,string nome, string endereco, long long cpf, long long telefone) : Pessoa(nome, endereco, cpf, telefone)
{
	this->assistente = assistente;
	this->agenda = agenda;
}

//construtor
Especialista::Especialista(string nome, string endereco, long long cpf, long long telefone) : Pessoa(nome, endereco, cpf, telefone)
{

}

//construtor
Especialista::Especialista(Especialista& especialista)
{
	setNome(especialista.getNome());
	setEndereco(especialista.getEndereco());
	setCpf(especialista.getCpf());
	setTelefone(especialista.getTelefone());

	this->assistente = assistente;
	this->agenda = agenda;

}

//destrutor
Especialista:: ~Especialista(){}


void Especialista::imprimirAgenda()//imprime toda a agenda de um Especialista
{
	cout << "Especialista: " << getNome() << " CPF: " << getCpf() << "\n\n"
		<< agenda;
}

//sobrecarga do operador de atribuicao '=' 

Especialista& Especialista:: operator=(Especialista& especialista)
{
	if (this == &especialista)
	{
		return *this;
	}

	setNome(especialista.getNome());
	setEndereco(especialista.getEndereco());
	setCpf(especialista.getCpf());
	setTelefone(especialista.getTelefone());


	this->assistente = assistente;
	this->agenda = agenda;

	
	return *this;
}


//sobrecarga do operador 'cout'
ostream& operator << (ostream& out, Especialista& especialista)
{
	out << "Nome: " << especialista.getNome() << "\n" << "Endereco: " << especialista.getEndereco() << "\n" << "Cpf: " << especialista.getCpf()
		<< "\n"
		<< "Telefone: " << especialista.getTelefone() << "\n";
	out << "\nAssistente Do Especialista:\n " << "Nome: " << especialista.assistente.getNome() << " CPF: " << especialista.assistente.getCpf();
	
	return out;
}